### ISSUE TITLE
Envoy crashes if proxy lib produces large record in access log

### ISSUE THREAD

**Issue by genbit** [cilium/cilium#9591]
We were working with Go extensions, with redis protocol, and found the following bug:

if redis proxylib produces long enough (17K symbols) messages in the envoy access log (see the attachment), envoy crashes with the following message:

```level=error msg="Error while parsing Envoy logs" error="bufio.Scanner: token too long" subsys=envoy-manager
level=debug msg="xDS stream canceled" error="rpc error: code = Canceled desc = context canceled" subsys=xds xdsStreamID=2
level=warning msg="Envoy: Proxy crashed" error="signal: broken pipe" subsys=envoy-manager
```

Presumable it has something to do with:
https://github.com/cilium/cilium/blob/master/pkg/envoy/envoy.go#L282

I was not able to reproduce it for HTTP protocols.

[scanner.log](https://github.com/cilium/cilium/files/3833320/scanner.log)

**Comment by genbit** [comment]
@jrajahalme did you recently fix something similar in context of TLS visibility?

**Comment by jrajahalme** [comment]
@genbit This was fixed by #9814, forgot to add a Fixes header in there.
